agent/engine/engine_integ_test.go

```diff
@@ -747,7 +747,7 @@ func TestPullContainerManifestInteg(t *testing.T) {
 				if tc.assertError == nil {
 					require.NoError(t, res.Error)
 					if tc.digestResolutionNotNeeded {
-						assert.Empty(t, container.GetImageDigest())
+						assert.False(t, container.DigestResolved())
 					} else {
```

Comment:
The `if` block can simply be deleted as the `else` block applies to both cases now. We should assert on `GetImageDigest()` for all cases. And by extension we can remove the `digestResolutionNotNeeded` field from the test cases.

Reply:
Thanks, updated